board-responsibilities.md

```diff
@@ -1,26 +1,43 @@
 ## Responsibilities of Board of Directors and Technical Steering Committee
 
-> [!WARNING]
-> In development, more details will be added soon.
+The NumFOCUS Board of Directors and the Technical Steering Committee are positions of authority.
+The former group deals with higher-level strategic and administrative decisions, while the latter deals with more hands-on day-to-day technical decisions.
+
+This document broadly lists the responsibilities of each group.
 
 ## Board of Directors - Administrative or Executive Branch
```

Comment:
Delete Executive Branch and use Administrative Board to be consistent with what the board is using.